linalg/solvers.hpp

```diff
@@ -473,6 +473,38 @@ class SLBQPOptimizer : public OptimizationSolver
    virtual void Mult(const Vector &xt, Vector &x) const;
 };
 
+class BlockILU0 : public Solver
+{
+public:
+   BlockILU0(int block_size_ = 1, bool reorder_ = true);
+
+   BlockILU0(Operator &op, int block_size_ = 1, bool reorder_ = true);
+
+   void SetOperator(const Operator &op);
+
+   void Mult(const Vector &b, Vector &x) const;
+
+   int *GetI() { return IB.GetData(); }
```

Comment:
Several things: (1) I think `GetBlockI()` would be a better name, (2) this public function should be documented at least a little, (3) unless it's only for testing, but in that case I think that fact should be documented. (ditto for `GetJ` and `GetData` methods below)

Reply:
Agreed re: naming. I'll do a pass and take care of the documentation.